jdaviz/app.py

```diff
@@ -955,6 +955,17 @@ def get_subsets(self, subset_name=None, spectral_only=False,
 
         all_subsets = {}
 
+        # If we only want one subset, no need to loop through them all
+        if subset_name is not None:
+            if isinstance(subset_name, str):
+                subsets = [subset for subset in subsets if subset.label == subset_name]
+                if subsets == []:
+                    all_labels = [sg.label for sg in dc.subset_groups]
+                    all_labels.sort()
```

Comment:
Save you one operation.
```suggestion
                    all_labels = sorted(sg.label for sg in dc.subset_groups)
```